package:
  name: cpp-exceptions-test
  version: "0.1"
  tag:
    - core
    - pyodide.test
    - library
source:
  path: src
build:
  type: shared_library
  script: |
    em++ ${SIDE_MODULE_CFLAGS} -fexceptions -I${PYTHONINCLUDE} -flto -fno-lto -std=c++20 -c throw.cpp -o throw.o
    em++ ${SIDE_MODULE_CFLAGS} -fexceptions -I${PYTHONINCLUDE} -flto -fno-lto -std=c++20 -c catch.cpp -o catch.o
    em++ ${SIDE_MODULE_LDFLAGS} -fexceptions  -o ${DISTDIR}/cpp-exceptions-test-throw.so throw.o
    em++ ${SIDE_MODULE_LDFLAGS} -fexceptions  -o ${DISTDIR}/cpp-exceptions-test-catch.so catch.o
